File: OpenMP/omp_d1_6_pro1_Q.c
#include <stdio.h>
// <<< !!

#define N 20

int main(void) {
    int A[N];

    #pragma omp parallel
    {
        int thread_id,num_threads;        
        int chunk_size;
        int start,end;

        thread_id   = 0; // <<< !!
        num_threads = 0; // <<< !!
        chunk_size  = 0; // <<< !!
        start       = 0; // <<< !!
        end         = 0; // <<< !!
        
        for (int i = 0; i < 0; i++) { // <<< !!
            A[i] = 100 + i;
        }
    }

    for (int i = 0; i < 20; i++) {
        printf("%8d %0.4d\n", i,A[i]);
    }
    
    return 0;
}

File: OpenMP/omp_d1_6_pro1_Q.f90
program main
    ! <<< !!
    implicit none

    integer, parameter :: N = 20
    integer :: A(0:N-1)

    integer :: i
    integer :: thread_id, num_threads
    integer :: chunk_size
    integer :: i_start, i_end

    !$omp parallel private(i,thread_id, num_threads, chunk_size, i_start, i_end)
        thread_id   = 0 ! <<< !!
        num_threads = 0 ! <<< !!
        chunk_size  = 0 ! <<< !!
        i_start     = 0 ! <<< !!
        i_end       = 0 ! <<< !!

        do i = 0, 0 ! <<< !!
            A(i) = 100 + i
        end do
    !$omp end parallel

    do i = 0, N-1
        write(*,'(I8,1X,I4.4)') i, A(i) 
    end do
end program main



